### PR TITLE
Pipeline support

### PR DESCRIPTION
**Pull request by hpsniper**
mabl can now be used as a pipeline step.

You can do this using full step syntax:
`step([$class: 'MablStepBuilder', restApiKey: 'jMzlz6eUkPYX4S5wOhOn5w', environmentId: 'WHVufN6h4SUWDVYWd8q17Q-e'])`

or using the syntax generator:
bamboo snippet generator

Resulting in the output:
`mabl applicationId: '6eRYl4zdcUbdmWlFh1aZTQ-a', environmentId: 'WHVufN6h4SUWDVYWd8q17Q-e', restApiKey: 'jMzlz6eUkPYX4S5wOhOn5w'`

Either can be used inside a node function for more context:
```
   stage('Test') {
       node {
           echo 'Hello Testing'
           step([$class: 'MablStepBuilder', restApiKey: 'jMzlz6eUkPYX4S5wOhOn5w', environmentId: 'WHVufN6h4SUWDVYWd8q17Q-e'])
       }
   }
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -21,13 +21,13 @@
...
-        <jenkins.version>1.580.1</jenkins.version>
-        <jenkins-test-harness.version>1.580.1</jenkins-test-harness.version>
+        <jenkins.version>1.651.2</jenkins.version>
+        <jenkins-test-harness.version>2.1</jenkins-test-harness.version>
         <release.skipTests>false</release.skipTests>
         <maven.javadoc.skip>true</maven.javadoc.skip>
         <findbugs-maven-plugin.version>3.0.2</findbugs-maven-plugin.version>
         <concurrency>1</concurrency>
-        <java.level>6</java.level>
+        <java.level>7</java.level>
```
Comment (hpsniper) [inline pom.xml]: Java 7 is required when using jenkins version >= 1.612

File: pom.xml [hunk 2]
```diff
@@ -21,13 +21,13 @@
 
     <properties>
         <!-- Last Jenkins LTS release -->
-        <jenkins.version>1.580.1</jenkins.version>
-        <jenkins-test-harness.version>1.580.1</jenkins-test-harness.version>
```
Comment (twistedpair) [inline pom.xml]: The reason for upgrading? Most similar plugins stick with the older version.

File: src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java [hunk 3]
```diff
@@ -40,27 +43,32 @@
...
 
     @DataBoundConstructor
     public MablStepBuilder(
             final String restApiKey,
             final String environmentId,
-            final String applicationId,
-            final boolean continueOnPlanFailure,
-            final boolean continueOnMablError
-
+            final String applicationId
```
Comment (twistedpair) [inline src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java]: Reason for removal from constructor? Is it so they are not required in the pipeline steps?

File: src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java [hunk 4]
```diff
@@ -77,11 +85,20 @@ public String getApplicationId() {
...
+    }
+
+    public boolean isContinueOnMablError() {
+        return this.continueOnMablError;
+    }
+
     @Override
-    public boolean perform(
-            final AbstractBuild<?, ?> build,
+    public void perform(
```
Comment (twistedpair) [inline src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java]: Doesn't match the overridden signature. Expects 

```java
void perform(@Nonnull Run<?,?> run, @Nonnull FilePath workspace, @Nonnull Launcher launcher,
                 @Nonnull TaskListener listener) throws InterruptedException, IOException;
```

File: src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java [hunk 5]
```diff
@@ -95,55 +112,47 @@ public boolean perform(
...
             outputStream.println("There was an execution error trying to run your journeys in mabl");
             e.printStackTrace(outputStream);
-            return continueOnMablError;
-
+            run.setResult(Result.FAILURE);
         } catch (TimeoutException e) {
             outputStream.printf("Oh dear. Your journeys exceeded the max plugin runtime limit of %d seconds.%n" +
                     "We've aborted this Jenkins step, but your journeys may still be running in mabl.", EXECUTION_TIMEOUT_SECONDS);
-            return continueOnMablError;
+            run.setResult(Result.FAILURE);
```
Comment (twistedpair) [inline src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java]: Are we removing the ability to continue on certain types of errors?

File: src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java (hunk 3 above)
Comment (hpsniper) [reply src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java]: That is correct.

File: src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java (hunk 5 above)
Comment (hpsniper) [reply src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java]: no, this was a mistake. Will fix

File: pom.xml (hunk 2 above)
Comment (hpsniper) [reply pom.xml]: Apparently, I don't need all the pipeline step dependencies that I thought I did. These versions can stay as they are.

**Pull request merged**